pkg/controller/azvolume.go

```diff
@@ -98,44 +97,52 @@ func (r *ReconcileAzVolume) Reconcile(ctx context.Context, request reconcile.Req
 
 		// if the GET failure is triggered by other errors, requeue the request
 		azVolume.Name = request.Name
-		return reconcileReturnOnError(azVolume, "get", err, r.retryInfo)
+		return reconcileReturnOnError(ctx, azVolume, "get", err, r.retryInfo)
 	}
 
+	ctx, _ = azureutils.GetWorkflowFromObj(ctx, azVolume)
+
 	// if underlying cloud operation already in process, skip until operation is completed
 	if isOperationInProcess(azVolume) {
-		klog.V(5).Infof("Another operation (%s) is already in process for the AzVolume (%s). Will be requeued once complete.", azVolume.Name, azVolume.Status.State)
 		return reconcileReturnOnSuccess(azVolume.Name, r.retryInfo)
 	}
 
 	// azVolume deletion
 	if objectDeletionRequested(azVolume) {
 		if err := r.triggerDelete(ctx, azVolume); err != nil {
 			//If delete failed, requeue request
-			return reconcileReturnOnError(azVolume, "delete", err, r.retryInfo)
+			return reconcileReturnOnError(ctx, azVolume, "delete", err, r.retryInfo)
 		}
 		//azVolume creation
 	} else if azVolume.Status.Detail == nil {
 		if err := r.triggerCreate(ctx, azVolume); err != nil {
-			klog.Errorf("failed to create volume (%s): %v", azVolume.Spec.VolumeName, err)
-			return reconcileReturnOnError(azVolume, "create", err, r.retryInfo)
+			return reconcileReturnOnError(ctx, azVolume, "create", err, r.retryInfo)
 		}
 		// azVolume update
 	} else if azVolume.Spec.CapacityRange != nil && azVolume.Spec.CapacityRange.RequiredBytes != azVolume.Status.Detail.CapacityBytes {
 		if err := r.triggerUpdate(ctx, azVolume); err != nil {
-			klog.Errorf("failed to update volume (%s): %v", azVolume.Spec.VolumeName, err)
-			return reconcileReturnOnError(azVolume, "update", err, r.retryInfo)
+			return reconcileReturnOnError(ctx, azVolume, "update", err, r.retryInfo)
 		}
-		// azVolume released, so clean up replica Attachments (primary Attachment should be deleted via UnpublishVolume request to avoid race condition)
 	}
 
 	return reconcileReturnOnSuccess(azVolume.Name, r.retryInfo)
 }
 
 func (r *ReconcileAzVolume) triggerCreate(ctx context.Context, azVolume *diskv1beta1.AzVolume) error {
+	var err error
+	var waitForGoroutine bool
+	ctx, workflow := azureutils.NewWorkflow(ctx)
```

Comment:
Similar comment as before, Should we pass AzVolume name as details param to NewWorkflow so that we can search the logs for that specific volume?

Reply:
We add the volume name and pv name to workflow details at the beginning of reconciliation extracting them from the CRI. So our workflow should contain the volume name in every logs it outputs.
https://github.com/sunpa93/azuredisk-csi-driver/blob/be8c10064ae1de99e92c17d5bff6a12a7483a934/pkg/azureutils/workflow.go#L51

Reply:
Makes sense. Since we are creating new worklfow from the object in this function, I wonder if it would be more clear to name it NewWorkflowFromObj  instead?

Reply:
Actually, we are creating a child workflow of the root workflow that has been stored to context in `Reconcile` function.